### PR TITLE
ui: login view: hide button in sso flow, tweak "login" vs. "log in"

### PR DESCRIPTION
**Pull request by jgehrcke**
This improves upon https://github.com/opstrace/opstrace/issues/243

### CODE REVIEW

**Comment by jgehrcke** [comment]
gcp ci failed as of https://github.com/opstrace/opstrace/issues/62.

**Comment by jgehrcke** [comment]
When we look at uishot-rootpage.png in the build artifacts then we see that the patch didn't take effect for the build. That's interesting.

```
[2021-01-06T16:55:56Z]    "prometheusOperator": "quay.io/prometheus-operator/prometheus-operator:v0.42.1",
[2021-01-06T16:55:56Z] -  "graphqlEngine": "opstrace/graphql:7232833b5b41ed1793caff32ca128926",
[2021-01-06T16:55:56Z] -  "app": "opstrace/app:7232833b5b41ed1793caff32ca128926",
[2021-01-06T16:55:56Z] +  "graphqlEngine": "opstrace/graphql:f79e575fd98938e009b5fc5f2f706b65",
[2021-01-06T16:55:56Z] +  "app": "opstrace/app:f79e575fd98938e009b5fc5f2f706b65",
[2021-01-06T16:55:56Z]    "redis": "bitnami/redis:6.0.9"
```

CC @MatApple @sreis we need to watch out for this.

**Comment by sreis** [comment]
Yeah, something went wrong since it deployed the app with the old image.
```
    Image:         opstrace/app:7232833b5b41ed1793caff32ca128926
    Image ID:      docker-pullable://opstrace/app@sha256:b6b11e3939b628d9041477095a991e5b2c56e17605d80ba17d0d3dc84058342e
```

**Comment by sreis** [comment]
Opened https://github.com/opstrace/opstrace/pull/248 to try and fix it.

File: packages/app/src/client/views/login/index.tsx [hunk 1]
```diff
@@ -148,14 +147,21 @@ const Login = (props: { state?: State }) => {
...
-  // second, create a session within the cluster, using the accessToken
+  // Use the access token to log in to (create a session with) the cluster.
   useEffect(() => {
     if (!accessToken || !user) {
       return;
     }
     (async function createSession() {
       try {
-        // create a session
+        // Login. Note(JP): the data in the body looks suspicious. I would
```
Comment (MatApple) [inline packages/app/src/client/views/login/index.tsx:157]: +1 to using /userinfo on the server instead!

File: packages/app/src/client/views/login/index.tsx [hunk 2]
```diff
@@ -239,6 +244,17 @@ const Login = (props: { state?: State }) => {
...
 
+  // Inspect query parameters: try to identify when we're in the middle of an
+  // SSO flow (when we're being redirected back from the identity provider) and
+  // do not display an interactive page in that case: no button, etc.
+  const querystring = window.location.search.substring(1);
+  const queryparms = new URLSearchParams(querystring);
+  const authzcode: null | string = queryparms.get("code")
+  if (authzcode !== null && authzcode.length > 1) {
+    // To render nothing, return null.
+    return null
```
Comment (MatApple) [inline packages/app/src/client/views/login/index.tsx:255]: Perhaps we should at least provide a spinner or some sort of feedback that the UI is working on something

Comment (jgehrcke) [reply packages/app/src/client/views/login/index.tsx:255]: yes!

**Pull request merged**